packages/components/pie-button/test/component/pie-button.spec.ts

```diff
@@ -272,422 +156,228 @@ test.describe('Form Actions', () => {
-                    document.body.appendChild(span);
-                });
+                // Assert
+                await expect(formIntegrationPage.formSubmittedFlag).toHaveCount(testCase.expectedFormSubmittedFlagCount);
             });
-
-            // Fill out the form
-            await page.fill('#username', 'testUser');
-            await page.fill('#password', 'testPassword');
-
-            // Act
-            // Press Enter in the password field
-            await page.focus('#password');
-            await page.press('#password', 'Enter');
-
-            // Assert
-            const formSubmittedFlagExists = Boolean(await page.$('#formSubmittedFlag'));
-            expect(formSubmittedFlagExists).toBe(false);
         });
 
         test('should NOT submit the form when pressing Enter on a pie-button that is not type of submit', async ({ page }) => {
             // Arrange
-            await page.evaluate(() => {
-                const formHTML = `
-                    <form id="testForm" action="/foo" method="POST">
-                        <input type="text" id="username" name="username" required>
-                        <input type="password" id="password" name="password" required>
-                        <pie-button type="submit">Submit</pie-button>
-                        <pie-button id="resetPieButton" type="reset">Reset</pie-button>
-                    </form>
-                `;
-                document.body.innerHTML = formHTML;
-
-                const form = document.querySelector('#testForm') as HTMLFormElement;
-
-                form.addEventListener('submit', (e) => {
-                    e.preventDefault();
-
-                    const span = document.createElement('span');
-                    span.id = 'formSubmittedFlag';
-                    document.body.appendChild(span);
-                });
-            });
+            const formIntegrationPage = new FormIntegrationPage(page, 'button--form-integration');
+            await formIntegrationPage.load();
 
             // Fill out the form
-            await page.fill('#username', 'testUser');
-            await page.fill('#password', 'testPassword');
+            await formIntegrationPage.fillForm(formInput);
 
             // Act
-            // Press Tab until we focus the reset button
-            await page.keyboard.press('Tab');
             await page.keyboard.press('Tab');
-
-            // Press Enter on the pie-button with type reset
-            await page.press('#resetPieButton', 'Enter');
+            await page.keyboard.press('Enter');
 
             // Assert
-            const formSubmittedFlagExists = Boolean(await page.$('#formSubmittedFlag'));
-            expect(formSubmittedFlagExists).toBe(false);
+            await expect(formIntegrationPage.formSubmittedFlag).toHaveCount(0);
         });
 
         test('should NOT submit the form when pressing Enter on a native non-submit button', async ({ page }) => {
```

Comment:
Pretty sure I wrote the original test here but I cannot figure out why we're testing this 🤔

Reply:
Yeah I think there's quite a few tests like this we probably don't need. I'll leave for now and do this as part of the wider audit.